### DISCUSSION TITLE
The whdload archive can be any filename you want (extension remains the same)

### DISCUSSION THREAD

**Discussion by giantclambake** [BlitterStudio/amiberry#1109] [General]
If I read through the wiki (and I've had others do likewise), the impression is given that if one downloads, for example here again AddamsFamily_v1.3_0419.lha , that the filename must remain as is...ie; 'AddamsFamily_v1.3_0419' [.lha]

Picture if you will, a linux desktop environment for children ~ in the file-manager, the above file appears as an icon by the name of AddamsFamily_v1.3_0419 (the file extension is hidden from/unchangeable to the user).

User decides to give the file a name they can relate to ~ RMB -> Rename and the filename becomes 'The Addams Family'[.lha]

The user then wants to launch the game ~   RMB -> Run with amiberry (invokes amiberry --autoload %F )

```
software perspective:

amiberry --autoload The Addams Family.lha

...from amiberry.log;

WHDLoad... /home/amiga/                        The Addams Family.lha
WHDBooter Launched
internal savedatapath [/home/amiga/Amiberry/whdboot/save-data]
SymLink Kickstart ROMs for Booter
internal savedatapath [/home/amiga/Amiberry/whdboot/save-data]
WHDBoot - using kickstarts from /home/amiga/Amiberry/whdboot/save-data/Kickstarts
Making SymLink for Kickstart ROM: /home/amiga/Amiberry/whdboot/save-data/Kickstarts/kick33180.A500  [Ok]
Making SymLink for Kickstart ROM: /home/amiga/Amiberry/whdboot/save-data/Kickstarts/kick37175.A500  [Ok]
Making SymLink for Kickstart ROM: /home/amiga/Amiberry/whdboot/save-data/Kickstarts/kick39106.A1200  [Ok]
Making SymLink for Kickstart ROM: /home/amiga/Amiberry/whdboot/save-data/Kickstarts/kick40063.A600  [Ok]
Making SymLink for Kickstart ROM: /home/amiga/Amiberry/whdboot/save-data/Kickstarts/kick40068.A1200  [Ok]
Making SymLink for Kickstart ROM: /home/amiga/Amiberry/whdboot/save-data/Kickstarts/kick40068.A4000  [Ok]
WHDBooter - Host: A600 ROM Available
WHDBooter - Loading whdload_db.xml
WHDBooter - Searching whdload_db.xml for                         The Addams Family
WHDBooter - Game H/W Settings:

                        PRIMARY_CONTROL=JOYSTICK
                        PORT0=JOY
                        PORT1=JOY
                        SCREEN_AUTOHEIGHT=TRUE
                        SCREEN_CENTERH=SMART
                        SCREEN_CENTERV=SMART
                        
WHDBooter - Default Slave: AddamsFamily.Slave
WHDBooter - SubPath:  AddamsFamily
WHDBooter - Libraries:  AddamsFamily
WHDBooter - Created Startup-Sequence
..
..
```

Obviously, what's happened here is the WHDBooter process found a match in the XML (using the AddamsFamily_v1.3_0419 definitions in the XML) and started/run the emulation & game as it should. It's not searching for 'The Addams Family' in a literal sense, it's searching the contents of the archive to try and match the XML entries.

Truth is, the WHDBooter process is filename agnostic ...for example;

mv AddamsFamily_v1.3_0419.lha wordassociationfootball.lha
amiberry --autoload wordassociationfootball.lha

Starts & runs as above..(caveat ' WHDBooter - Searching whdload_db.xml for        wordassociationfootball '  =)

This was another motivator to request 'an option to save a config based on the loaded title (either floppy or WHDLoad) in the Quickstart panel' (see #1101). Desktop users will change icon/file names, and who can blame them when you've got a filename like 'Dick Tracy (1990)(Titus)(M5)[cr Angels - Defjam][t +3 LGD - Masters].zip' that can become 'Dick Tracy.zip'  =)

It of course relies on the user to rename the file before starting amiberry and saving a config for that titlename, but IRL practice this pretty much takes care of itself...ie; user perspective/prerogative ...if you download  'Addams Family, The (1992)(Ocean)[cr FSN][t +2 SCX].zip' ...and see that as an icon name, it looks just bad/ugly enough to prompt most users into renaming it to something human readable like 'The Addams Family.zip' ...or 'Gomez&Morticia.zip' ...whatever you want.

How did I get here? Probably due to an understatement in   https://github.com/BlitterStudio/amiberry/wiki/WHDLoad-Auto-booting#how-it-works

"_Amiberry will automatically mount the HD booter and the game data to the emulated system for easy loading. It will also attempt to select appropriate settings for the game, based on the information available._"

Quick fix;

Amiberry will automatically mount the HD booter and the game data to the emulated system for easy loading. It will also attempt to select appropriate settings for the game from the XML file, based on the information available within the whdload archive.

HTH

**Comment by midwan** [comment]
The first attempt is to match the filename - if that fails, then it will do an attempt to match the SHA1 checksum value of the archive, with those that exist in the XML.

**Comment by giantclambake** [comment]
Yes, you & I know this, but the distinction is lost wrt younger/non-savvy users ~ they can't grasp that renaming the container does not change the container's size (for want of a better explanation =)

I'm approaching this incorrectly ~ the solution for me is to clone the wiki and update that text for my user audience (included as a documentation/help link in browser bookmarks) ; problem solved  ;)

**Reply by dimitris-norce** [reply]
I've updated that section of the wiki :)
